### PR TITLE
[IIDM v1.2] Add version 1.2.0

### PR DESCRIPTION
**Pull request by sebalaig**
Add version 1.2.0 and prepare/update tests for this new version.

Signed-off-by: Sébastien LAIGRE <slaigre@silicom.fr>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
#121 

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature: add the new version 1.2.0

**What is the current behavior?** *(You can also link to an open issue here)*
Current and last version is V1.1.0.

**What is the new behavior (if this is a feature change)?**
Current and last version is V1.2.0.
Roadtrip tests for this new version are available.
Roadtrip ref for extensions have been updated. 

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*
No

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: test/iidm/converter/xml/extensions/LoadMockExtXmlSerializer.cpp [hunk 1]
```diff
@@ -29,6 +29,7 @@ LoadMockExtXmlSerializer::LoadMockExtXmlSerializer() :
         VersionsCompatibilityBuilder()
             .put(IidmXmlVersion::V1_0(), {"1.0"})
             .put(IidmXmlVersion::V1_1(), {"1.1", "1.2"})
+            .put(IidmXmlVersion::V1_2(), {"1.2"})
```
Comment (sebalaig) [inline test/iidm/converter/xml/extensions/LoadMockExtXmlSerializer.cpp:32]: Don't know how it works in java without adding this version in the map, unless it throws an exception when calling `LoadMockExtXmlSerializer::getVersion()`...

Comment (mathbagu) [reply test/iidm/converter/xml/extensions/LoadMockExtXmlSerializer.cpp:32]: It's not allowed to have `1.2` compatible with XIIDM v1.1 and v1.2
In Java, this class is used only to check the compatibility support. The test runs only in v1.1. Maybe you should adapt the test.

File: test/resources/V1_1/eurostag-tutorial-example1-with-terminalMock-ext.xml [hunk 2]
```diff
@@ -1,44 +1,44 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<iidm:network xmlns:iidm="http://www.powsybl.org/schema/iidm/1_1" xmlns:mock="http://www.powsybl.org/schema/iidm/ext/terminal_mock/1_1" id="sim1" caseDate="2013-01-15T18:45:00.000+01:00" forecastDistance="0" sourceFormat="test">
+<iidm:network xmlns:iidm="http://www.powsybl.org/schema/iidm/1_1" xmlns:mock="http://www.powsybl.org/schema/iidm/ext/terminal_mock/1_1" id="sim1" caseDate="2013-01-15T18:45:00+01:00" forecastDistance="0" sourceFormat="test">
```
Comment (mathbagu) [inline test/resources/V1_1/eurostag-tutorial-example1-with-terminalMock-ext.xml:2]: Not sure to understand why this file has to be modified.

Comment (sebalaig) [reply test/resources/V1_1/eurostag-tutorial-example1-with-terminalMock-ext.xml:2]: This file (V1_1/eurostag-tutorial-example1-with-terminalMock-ext.xml) was erroneous but was not read as the concerning test was executed for every previous versions (i.e only for V1.0) : 
```
BOOST_AUTO_TEST_CASE(TerminalExtension) {
    ....
    test::converter::RoundTrip::roundTripAllPreviousVersionedXmlTest("eurostag-tutorial-example1-with-terminalMock-ext.xml");
}
```
=> updated

Comment (mathbagu) [reply test/resources/V1_1/eurostag-tutorial-example1-with-terminalMock-ext.xml:2]: In the Java code, this file is used in all versions:
```
    @Test
    public void testTerminalExtension() throws IOException {
        Network network2 = roundTripXmlTest(EurostagTutorialExample1Factory.createWithTerminalMockExt(),
                NetworkXml::writeAndValidate,
                NetworkXml::read,
                getVersionedNetworkPath("eurostag-tutorial-example1-with-terminalMock-ext.xml", CURRENT_IIDM_XML_VERSION));
        Load loadXml = network2.getLoad("LOAD");
        TerminalMockExt terminalMockExtXml = loadXml.getExtension(TerminalMockExt.class);
        assertNotNull(terminalMockExtXml);
        assertSame(loadXml.getTerminal(), terminalMockExtXml.getTerminal());

        // backward compatibility
        roundTripAllPreviousVersionedXmlTest("eurostag-tutorial-example1-with-terminalMock-ext.xml");
    }

```

File: test/iidm/converter/xml/OptionalLoadTypeBugTest.cpp [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+namespace converter {
+
+namespace xml {
+
+BOOST_AUTO_TEST_SUITE(OptionalLoadTypeTest)
+
+BOOST_AUTO_TEST_CASE(OptionalLoadTypeFromXml) {
+    Network network10 = Network::readXml(test::converter::RoundTrip::getVersionedNetwork("optionalLoadTypeBug.xml", IidmXmlVersion::V1_0()));
+    BOOST_CHECK_EQUAL(static_cast<int>(LoadType::UNDEFINED), static_cast<int>(network10.getLoad("L").getLoadType()));
```
Comment (mathbagu) [inline test/iidm/converter/xml/OptionalLoadTypeBugTest.cpp:27]: Why do you use static cast? Is it possible to compare LoadType directly?

**Comment by mathbagu** [comment]
Fix #121

**Pull request merged**